### DISCUSSION TITLE
RFC: Include ctx in getStaticProps

### DISCUSSION THREAD

**Discussion by birkir** [vercel/next.js#15458] [Ideas]
https://github.com/vercel/next.js/pull/15255

Hi everyone.

I have a use-case where I need to access the http request on SSG Previews.

It wouldn't hurt to pass the ctx as a prop on getStaticProps, so that's is my proposal.

Example usage would be:

```tsx
export function getStaticProps(props: GetStaticPropsContext) {
  if (props.preview) {
    const rawCookies = props.ctx?.req?.headers?.cookie;
    // do something with cookies.
  }
}
```

The cookie I am trying to read is set by an Iframe on a different domain that is rendered on the page.

It will change between routes, hence not able to use `setPreviewData` api.

**Comment by jkjustjoshing** [comment]
`getStaticProps` is only run at *build* time, not in response to a given request.

You might be looking for `getServerSideProps` or `getInitialProps`?

**Reply by birkir** [reply]
It is run in preview mode at runtime 😉

**Comment by birkir** [comment]
Paging @Timer 
Any feedback?

**Comment by birkir** [comment]
Its a very simple change that will benefit many use cases

https://github.com/vercel/next.js/pull/15255
